tokio/src/runtime/builder.rs

```diff
@@ -877,6 +881,29 @@ impl Builder {
         }
     }
 
+    cfg_metrics! {
+        pub fn metrics_poll_count_histogram(&mut self, histogram_scale: crate::runtime::HistogramScale) -> &mut Self {
```

Comment:
My preference here would be for a single function that takes either all necessary parameters or a builder or config struct.

I've had poor experiences in the past with public APIs where setting 1 of N values implicitly enables some functionality. If the poll count histogram is going to be opt-in, then I believe that a more self explanatory API would be a single function like `metrics_poll_count_histogram(&mtu self, config: PollCountHistogramConfig)`. You can then implement `Default` for the config struct to allow the user to modify only some values if that is desireable (which I think it is).

Reply:
Multiple arguments to a builder fn is not a forwards-compatible option, unfortunately. The same goes with a `Config` struct. If we make the struct `#[non_exaustive],` users cannot initialize it. That leaves nesting builders, which gets messy.

What issues have you had in the past?

Reply:
User confusion issues in both directions.
* How do I enable X function?
* I didn't realise that setting this value would enable X function.
* I didn't realise that *not* setting this value would disable X function.

Also, there's currently no way to "enable poll count histogram with default values", you always have to override one value to enable, even if you want to stick to the defaults.

My personal feeling is to go with nesting builders, because in the default case the user would pass `PollCountHistogramConfig::default()`, which I think isn't so bad.

Reply:
Of course, as you said on Discord, it's an unstable API, so trying it out as is isn't such an issue.

Reply:
Good points. To address those issues, I added an `enable_metrics_poll_count_histogram` builder fn. This is the only way to enable that metric. The other builder fns will not implicitly enable or disable the feature.